extensions/Worker.Extensions.Http.AspNetCore/src/FunctionsMiddleware/FunctionsHttpProxyingMiddleware.cs

```diff
@@ -49,26 +59,53 @@ public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next
 
             var invocationResult = context.GetInvocationResult();
 
-            if (invocationResult?.Value is IActionResult actionResult)
+            if (_httpResponseTypes.Any(type => invocationResult?.Value?.GetType() == type))
             {
-                ActionContext actionContext = new ActionContext(httpContext, httpContext.GetRouteData(), new ActionDescriptor());
-
-                await actionResult.ExecuteResultAsync(actionContext);
+                HandleHttpResult(invocationResult.Value, context, httpContext);
             }
-            else if (invocationResult?.Value is AspNetCoreHttpResponseData)
+            else
             {
-                // The AspNetCoreHttpResponseData implementation is
-                // simply a wrapper over the underlying HttpResponse and
-                // all APIs manipulate the request.
-                // There's no need to return this result as no additional
-                // processing is required.
-                invocationResult.Value = null;
+                HandleHttpResultInOutputBindings(context, httpContext);
             }
 
             // allows asp.net middleware to continue
             _coordinator.CompleteFunctionInvocation(invocationId);
         }
 
+        private static async void HandleHttpResult(object? httpResult, FunctionContext context, HttpContext httpContext)
+        {
+            switch (httpResult)
+            {
+                case HttpResponseData httpResponseData:
+                    context.GetInvocationResult().Value = null;
+                    break;
+
+                case IActionResult actionResult:
+                    ActionContext actionContext = new ActionContext(httpContext, httpContext.GetRouteData(), new ActionDescriptor());
+                    await actionResult.ExecuteResultAsync(actionContext);
+                    break;
+
+                case IResult iResult:
+                    await iResult.ExecuteAsync(httpContext);
+                    break;
+
+                default:
+                    throw new InvalidOperationException($"Unexpected return type {httpResult?.GetType()} for HTTP function {context.FunctionDefinition.Name}.");
+            }
+        }
+
+        private static void HandleHttpResultInOutputBindings(FunctionContext context, HttpContext httpContext)
+        {
+            var httpOutputBinding = context.GetOutputBindings<object>().FirstOrDefault(a => string.Equals(a.BindingType, HttpBindingType, StringComparison.OrdinalIgnoreCase));
+
+            if (httpOutputBinding is null)
+            {
+                throw new InvalidOperationException($"No HTTP Response type could be found in the function {context.FunctionDefinition.Name}'s output bindings.");
```

Comment:
Similar issue to the one with the new exception above, this shouldn't fail (it's not strictly required)